### ISSUE TITLE
Quirk in lib/lib.es6.d.ts

### ISSUE THREAD

**Issue by sebastian-lenz** [Microsoft/TypeScript#17107]
**TypeScript Version:**  2.4.1
**Code**

```ts
// "target": "es6"
const obj = {};
Object.defineProperty(obj, Symbol('foo'), {  // <-- Typescript error points here
  enumerable: true,
  value: null,
  get: null, // <-- This is the actual error
});
```

**Expected behavior:**
TypeScript should emit an error cause of the invalid third argument.

**Actual behavior:**
TypeScript emits an error cause of the second argument: `error TS2345: Argument of type 'symbol' is not assignable to parameter of type 'string'.`

**Cause of the problem:**
There are two definitions of `ObjectConstructor.defineProperty` in [lib/lib.es6.d.ts](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.es6.d.ts#L178), the first definition comes from [src/lib/es5.d.ts#L158](https://github.com/Microsoft/TypeScript/blob/master/src/lib/es5.d.ts#L158) and looks like this:
```ts
defineProperty(o: any, p: string, attributes: PropertyDescriptor & ThisType<any>): any;
```
The second definition comes from [src/lib/es2015.core.d.ts#L353](https://github.com/Microsoft/TypeScript/blob/master/src/lib/es2015.core.d.ts#L353) and looks like this:
```ts
defineProperty(o: any, propertyKey: PropertyKey, attributes: PropertyDescriptor): any;
```

As lib/lib.es6.d.ts is generated from the other two source files, I currently see no easy fix for this quirk. Changing the signature in src/lib/es5.d.ts to accept symbols would be incorrect as symbols are not allowed here. 

However, the roots of my struggle also can be found in src/lib/es5.d.ts as the decorator signatures are defined here and I ran into this quirk while playing around with them. The decorator signatures in [src/lib/es5.d.ts#L1249](https://github.com/Microsoft/TypeScript/blob/master/src/lib/es5.d.ts#L1249) contain the otherwise forbidden symbol type:
```ts
declare type PropertyDecorator = (target: Object, propertyKey: string | symbol) => void;
declare type MethodDecorator = <T>(target: Object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>) => TypedPropertyDescriptor<T> | void;
declare type ParameterDecorator = (target: Object, propertyKey: string | symbol, parameterIndex: number) => void;
```

**Conclusion:**
At first I thought I've ran into a simple declaration error, then I thought there is a serious flaw in the typing system and in the end I must realize it's a simple quirk caused by the way the definitions are created and the error was in front of the keyboard. Keep up the good work!

**Comment by mhegazy** [comment]
If overload resolution failed, the compiler repeats the process to reports errors. Overload resolution is order dependent, and so is error reporting if no overload was found. As you noted the fix is to change the order of the library, we are in the process of refactoring the library into smaller pieces that will address this issue. for now a workaround is to list `--lib`, e.g.: ` --lib dom,es5,es6` on the command line or `"lib": ["dom", "es5", "es6"]` in tsconfig.json; this will ensure that you get the overloads in the right order.
